src/main/java/selic/re/discordbridge/DiscordBot.java

```diff
@@ -125,16 +127,13 @@ public void onMessageReceived(MessageReceivedEvent event)
             if (!msg.getAttachments().isEmpty()) {
                 for (Message.Attachment attachment : msg.getAttachments()) {
                     root.append(" [");
-                    LiteralText text;
-                    if (attachment.isImage()) {
-                        text = new LiteralText("image");
-                    } else if (attachment.isVideo()) {
-                        text = new LiteralText("video");
-                    } else {
-                        text = new LiteralText("attachment");
-                    }
+                    // If the file is intended to be hidden by a spoiler, we wrap it with exclamation marks: [!file.txt!]
+                    LiteralText text = new LiteralText(FILE_SPOILER.matcher(attachment.getFileName()).replaceFirst("!$1!"));
                     ClickEvent click = new ClickEvent(ClickEvent.Action.OPEN_URL, attachment.getUrl());
-                    HoverEvent hover = new HoverEvent(HoverEvent.Action.SHOW_TEXT, new LiteralText(attachment.getFileName() + "\n" + (attachment.getSize() / 1000) + " kb"));
```

Comment:
imo this is less useful, the filename is usually long and I've never seen it helpful - it's usually "text.txt" or "unknown.png" or, if you're lucky, "IMG_20210724_143556015.jpg". Screen space is expensive, file type is useful and short.

Reply:
Should the message contain what I currently have in the tooltip?

`[text/plain | 103 bytes]`

And the tooltip could contain the file name?

Reply:
I'd argue not. You can see that when you go to click on it because of the tool tip, and having it there doesn't influence your decision to want to click it - just extra cruft imo.

All users will care about is that there's a thing to click on, and we give them a vague description of what to expect - image, video, attachment. Maybe for the attachment case you can add the suffix ('.txt attachment') but mime type is highly technical and not useful for the majority of players.

Reply:
Oh right, fair point, I think I was looking at it from a poor perspective. The current implementation is fine then I think.